src/nitypes/waveform/_extended_properties.py

```diff
@@ -19,14 +23,24 @@ class ExtendedPropertyDictionary(MutableMapping[str, ExtendedPropertyValue]):
         over the network or write it to a TDMS file.
     """
 
-    __slots__ = ["_properties"]
+    __slots__ = ["_properties", "_on_key_changed"]
 
     def __init__(self, properties: Mapping[str, ExtendedPropertyValue] | None = None, /) -> None:
         """Initialize a new ExtendedPropertyDictionary."""
         self._properties: dict[str, ExtendedPropertyValue] = {}
+        self._on_key_changed: OnKeyChangedCallback | None = None
         if properties is not None:
             self._properties.update(properties)
 
+    @property
+    def on_key_changed(self) -> OnKeyChangedCallback | None:
```

Comment:
Let's keep this private rather than add it to the public API. Deleting the property and changing DigitalWaveform to set `_on_key_changed` directly would be fine.